Remote id fronted: hide all indicators if RID comms is not healthy:

Davidsastresas opened the pull request:
All this remote id feature depends on a device being detected so it makes sense to not show the rest of the flags until a healthy connection has been detected. Now it looks like this until we have healthy connection with RID device ( arm status message received ):

![rid settigns](https://github.com/mavlink/qgroundcontrol/assets/18221398/9e7eb1d2-d5dc-4764-96e3-ce081a205047)
![rid indicator](https://github.com/mavlink/qgroundcontrol/assets/18221398/f41b2581-6871-4b1e-9cfa-eeba2445782a)

It addresses the issues mentioned here #10654.

booo commented:
Does it make sens to communicate by text what is happening? I get that red is "danger" or "something is broken" but I don't get that a device is missing from that.

Davidsastresas commented:
@booo It  makes a lot of sense, I hadn't thought about it. I modified it so now when there is no connection, the label shows "NOT CONNECTED" instead of "RID COMMS" just with red background. It looks as follows:

![rid3](https://github.com/mavlink/qgroundcontrol/assets/18221398/138a4310-3944-4aef-810d-1eccc9824bb0)
![rid2](https://github.com/mavlink/qgroundcontrol/assets/18221398/26a6a375-ec09-447b-ac58-15cbb64166c3)

I squashed and extended the commit comment. Let me know if this looks good now. Thanks for the review!

booo commented:
Looks good now.

BluemarkInnovations commented:
I tested it with both ArduPilot and PX4. Looks indeed good.
